Trimble.Modus.Components/Controls/TMNumberInput/TMNumberInput.xaml.cs

```diff
@@ -0,0 +1,231 @@
+    /// Gets or Sets Max value for the input
+    /// </summary>
+    public double MaxValue
+    {
+        get { return (double)GetValue(MaxValueProperty); }
+        set { SetValue(MaxValueProperty, value); }
+    }
+    /// <summary>
+    /// Gets or Sets Min value for the input
+    /// </summary>
+    public double MinValue
+    {
+        get { return (double)GetValue(MinValueProperty); }
+        set { SetValue(MinValueProperty, value); }
+    }
+    /// <summary>
+    /// Gets or Sets value for the input
+    /// </summary>
+    public double Value
+    {
+        get { return string.IsNullOrEmpty(TMInputControl.Text) ? double.NaN : double.Parse(TMInputControl.Text); }
+        set { SetValue(ValueProperty, value); }
+    }
+    /// <summary>
+    /// Gets or Sets Step value for the input
+    /// </summary>
+    public double Step
+    {
+        get { return (double)GetValue(StepProperty); }
+        set { SetValue(StepProperty, value); }
+    }
+    /// <summary>
+    /// Gets or Sets IsEnabled value for the input
+    /// </summary>
+    public new bool IsEnabled
+    {
+        get { return (bool)GetValue(IsEnabledProperty); }
+        set { SetValue(IsEnabledProperty, value); }
+    }
+
+    /// <summary>
+    /// Gets or Sets IsReadOnly value for the input
+    /// </summary>
+    public bool IsReadOnly
+    {
+        get { return (bool)GetValue(IsReadOnlyProperty); }
+        set { SetValue(IsReadOnlyProperty, value); }
+    }
+
+    #endregion
+    /// <summary>
+    /// Initializes a new instance of the <see cref="TMNumberInput"/> class.
+    /// </summary>
+    public TMNumberInput()
+    {
+        InitializeComponent();
+        BindingContext = this;
+        TMInputControl.SetCenterTextAlignment();
+        TMInputControl.RightIconCommand = new Command(PlusCommand.Execute);
+        TMInputControl.LeftIconCommand = new Command(MinusCommand.Execute);
+    }
+
+    #region Private methods
+    /// <summary>
+    /// Called when the <see cref="MinValue"/> property is changed.
+    /// </summary>
+    /// <param name="bindable">The object that the property was changed on.</param>
+    /// <param name="oldValue">The old value of the property.</param>
+    /// <param name="newValue">The new value of the property.</param>
+    /// <exception cref="ArgumentException">Thrown when the new minimum value is greater than the maximum value.</exception>
+    private static void OnMinValuePropertyChanged(BindableObject bindable, object oldValue, object newValue)
+    {
+        TMNumberInput numberInput = bindable as TMNumberInput;
+        double min = (double)newValue;
+        double max = numberInput.MaxValue;
+        if (min > max)
+        {
+            throw new ArgumentException("Min value cannot be greater than max value");
+        }
+        else if (numberInput.Value < min)
+        {
+            numberInput.Value = min;
+        }
+        numberInput.TMInputControl.ToggleRightIconState(numberInput.Value > min);
```

Comment:
In both `OnMinValuePropertyChanged` and `OnMaxValuePropertyChanged`, you are toggleing the Right icon. Is that right?

Reply:
No, updated